File: app/c/server/winlibs/code/src/manager/GProcess.c
//===============================================
#include "GProcess.h"
#include "GTest.h"
#include "GSocket.h"
//===============================================
static void GProcess_delete(GProcess* _this);
static void GProcess_init(GProcess* _this);
static void GProcess_clean(GProcess* _this);
static void GProcess_run(GProcess* _this, int _argc, char** _argv);
static void GProcess_runTest(GProcess* _this, int _argc, char** _argv);
static void GProcess_runServer(GProcess* _this, int _argc, char** _argv);
//===============================================
GProcess* GProcess_new() {
    GProcess* lObj = (GProcess*)malloc(sizeof(GProcess));
    lObj->m_obj = GObject_new();

    lObj->delete = GProcess_delete;
    lObj->init = GProcess_init;
    lObj->clean = GProcess_clean;
    lObj->run = GProcess_run;
    lObj->runTest = GProcess_runTest;
    lObj->runServer = GProcess_runServer;
    return lObj;
}
//===============================================
static void GProcess_delete(GProcess* _this) {
    assert(_this);
    _this->m_obj->delete(_this->m_obj);
    free(_this);
}
//===============================================
static void GProcess_init(GProcess* _this) {
    assert(_this);
    SetConsoleOutputCP(CP_UTF8);
    xmlInitParser();
    xmlKeepBlanksDefault(0);
}
//===============================================
static void GProcess_clean(GProcess* _this) {
    assert(_this);
    xmlCleanupParser();
    xmlMemoryDump();
    GFunctions_delete();
}
//===============================================
static void GProcess_run(GProcess* _this, int _argc, char** _argv) {
    assert(_this);
    GLog* lLog = _this->m_obj->m_logs;
    const char* lModule = "";
    if(_argc > 1) lModule = _argv[1];

    if(!strcmp(lModule, "")) {
        lLog->addError(lLog, "Le module est obligatoire.");
    }
    else if(!strcmp(lModule, "test")) {
        _this->runTest(_this, _argc, _argv);
    }
    else if(!strcmp(lModule, "server")) {
        _this->runServer(_this, _argc, _argv);
    }
    else {
        lLog->addError(lLog, "Le module est inconnu.");
    }

    lLog->showLogsX(lLog);
}
//===============================================
static void GProcess_runTest(GProcess* _this, int _argc, char** _argv) {
    assert(_this);
    GLog* lLog = _this->m_obj->m_logs;
    GTest* lObj = GTest_new();
    lObj->run(lObj, _argc, _argv);
    lLog->addLogs(lLog, lObj->m_obj->m_logs);
    lObj->delete(lObj);
}
//===============================================
static void GProcess_runServer(GProcess* _this, int _argc, char** _argv) {
    assert(_this);
    GLog* lLog = _this->m_obj->m_logs;
    GSocket* lObj = GSocket_new();
    lObj->run(lObj, _argc, _argv);
    lLog->addLogs(lLog, lObj->m_parent->m_logs);
    lObj->delete(lObj);
}
//===============================================

File: app/python/server/code/src/manager/GProcess.py
#================================================
from functions import *
from manager.GObject import GObject
from manager.GTest import GTest
from manager.GSocket import GSocket
#================================================
class GProcess(GObject):
    #================================================
    def __init__(self):
        GObject.__init__(self)
    #================================================
    def init(self):
        pass
    #================================================
    def clean(self):
        pass
    #================================================
    def run(self):
        lModule = ""
        if len(sys.argv) > 1:
            lModule = sys.argv[1]
            
        if lModule == "":
            self.m_logs.addError("Le module est obligatoire.");
        elif lModule == "test":
            self.runTest();
        elif lModule == "server":
            self.runServer();
        else:
            self.m_logs.addError("Le module est inconnu.");
    #================================================
    def runTest(self):
        lObj = GTest()
        lObj.run()
        self.m_logs.addLogs(lObj.m_logs)
    #================================================
    def runServer(self):
        lObj = GSocket()
        lObj.runServer()
        self.m_logs.addLogs(lObj.m_logs)
#================================================
